tests/lib/rules/key-spacing.js

```diff
@@ -499,6 +499,15 @@ ruleTester.run("key-spacing", rule, {
                 mode: "minimum"
             }
         }]
+    },
+
+    // https://github.com/eslint/eslint/issues/5724
+    {
+        code: "({...object})",
```

Comment:
Can you add an invalid test similar to this for `({ a:b, ...object, c : d })` (still `align: "colon"`) with failures for the missing space before `b` and the extra space after `c` as a sanity check to make sure we don't accidentally break this in the future?